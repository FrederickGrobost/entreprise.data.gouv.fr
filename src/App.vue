<template>
  <div id="app">
    <navbar></navbar>
    <div class="notification full-width">
      Ce site est un travail en cours, actuellement en beta. Vous pouvez le
      consulter librement.
    </div>
    <search></search>
    <!-- TODO: put home in router-view (maybe?) -->
    <home v-if="showWelcomeText"></home>
    <router-view></router-view>
    <footer-etalab></footer-etalab>
  </div>
</template>

<script>
import Navbar from "@/components/Navbar";
import Search from "@/components/Search";
import Home from "@/components/Home";
// import Results from "@/components/Results";
// import Etablissement from "@/components/Etablissement";
import FooterEtalab from "@/components/FooterEtalab";

export default {
  name: "app",
  metaInfo: {
    title: null,
    titleTemplate: "%s | Entreprise.data.gouv.fr"
  },
  components: {
    Navbar: Navbar,
    Search: Search,
    Home: Home,
    // Results: Results,
    // Etablissement: Etablissement,
    FooterEtalab: FooterEtalab
  },
  computed: {
    showWelcomeText() {
      return this.$store.getters.isWelcomeTextVisible;
    }
  }
};
</script>

<style lang="scss" scoped>
#app {
  min-height: 100vh;
}
</style>
