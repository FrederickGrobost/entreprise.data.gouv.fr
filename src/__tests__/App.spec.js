import App from "@/App.vue";
import { createLocalVue, mount, RouterLinkStub } from "@vue/test-utils";
import { __createMocks as createStoreMocks } from "@/store/index.js";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.component("router-link", RouterLinkStub);
jest.mock("@/store/index.js");
jest.mock("mapbox-gl", () => jest.fn());

describe("App.vue", () => {
  const storeMocks = createStoreMocks();
  const $route = {
    params: {
      siret: "83305720100018",
      searchId: "mock-searchId"
    },
    path: "mock-path",
    query: {
      page: "mock-page"
    }
  };

  const wrapperApp = mount(App, {
    localVue,
    store: storeMocks.store,
    mocks: { $route },
    stubs: ["router-view"]
  });

  test("It match the snapshot", () => {
    expect(wrapperApp.vm.$el).toMatchSnapshot();
  });
});
