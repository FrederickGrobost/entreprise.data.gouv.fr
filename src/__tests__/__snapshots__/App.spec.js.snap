// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue It match the snapshot 1`] = `
<div
  id="app"
>
  <header
    class="navbar"
    role="navigation"
  >
    <div
      class="navbar__container"
    >
      <a
        class="navbar__home"
      >
        <img
          alt="entreprise.data.gouv.fr"
          class="navbar__logo"
          src="../assets/img/entreprise.svg"
        />
      </a>
    </div>
  </header>
   
  <div
    class="notification full-width"
  >
    
    Ce site est un travail en cours, actuellement en beta. Vous pouvez le
    consulter librement.
  
  </div>
   
  <div
    class="hero"
    role="banner"
  >
    <div
      class="hero__container container"
    >
      <div
        class="text-center"
        name="fade"
      >
        <h1
          class="search__title"
        >
          
          Retrouvez toutes les informations publiques concernant les
          entreprises et associations de France
        
        </h1>
         
        <p
          class="search__subtitle"
        >
          
          Les bases de données sur l’état civil des entreprises et
          associations françaises sont maintenant accessibles à tous, sans
          frais.
        
        </p>
      </div>
       
      <div
        class="form__group"
      >
        <input
          name="search"
          placeholder="Nom, SIREN, SIRET, adresse..."
          type="text"
        />
         
        <button
          class="overlay-button"
        >
          <svg
            class="icon icon-search"
          >
            <use
              xlink:href="#icon-search"
            />
          </svg>
        </button>
         
        <ul
          class="suggestions"
          style="display: none;"
        />
      </div>
       
      <!---->
    </div>
  </div>
   
  <div>
    <section
      class="api text-center section"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="column"
          >
            <p>
              
          Vous pouvez, si vous le souhaitez, interroger directement les APIs
        
            </p>
             
            <a
              class="button"
            >
              Documentation des APIs
            </a>
          </div>
           
          <div
            class="column api__image"
          >
            <img
              alt="man-at-desktop"
              src="../../assets/img/man-at-desktop.svg"
              width="200"
            />
          </div>
           
          <div
            class="column"
          >
            <p>
              Téléchargez les données de référence sur le site data.gouv.fr
            </p>
             
            <a
              class="button"
              href="https://www.data.gouv.fr/fr/reference"
            >
              Données de référence
            </a>
          </div>
        </div>
      </div>
    </section>
     
    <section
      class="section section-grey text-center public-administration"
    >
      <div
        class="container"
      >
        <p>
          
      Vous êtes une administration ?
          <br />
          
      Demandez un accès spécial à l'API Entreprise et obtenez des informations
      supplémentaires sur les sociétés
    
        </p>
         
        <a
          class="button"
          href="https://entreprise.api.gouv.fr/"
        >
          API Entreprise
        </a>
      </div>
    </section>
     
    <form
      class="text-center"
    >
      <h4>
        Inscription Mailing-List
      </h4>
       
      <p>
        
    Vos données sont traitées par Mailjet.
        <br />
        
    Vos informations ne seront jamais utilisées à des fins commerciales.
  
      </p>
       
      <iframe
        frameborder="0"
        height="95"
        marginheight="0"
        marginwidth="0"
        scrolling="no"
        src="https://app.mailjet.com/widget/iframe/zR1/ack"
        width="340"
      />
    </form>
  </div>
   
  <router-view-stub />
   
  <footer
    class="footer"
  >
    <div
      class="container"
    >
      <div
        class="footer__logo"
      >
        <a
          href="https://www.etalab.gouv.fr"
          target="_blank"
        >
          <img
            src="../assets/img/etalab.svg"
          />
        </a>
         
        <ul
          class="footer__social"
        >
          <li>
            <a
              href="https://www.twitter.com/etalab"
            >
              <svg
                class="icon icon-twitter"
              >
                <use
                  xlink:href="#icon-twitter"
                />
              </svg>
            </a>
          </li>
           
          <li>
            <a
              href="https://www.github.com/etalab"
            >
              <svg
                class="icon icon-github"
              >
                <use
                  xlink:href="#icon-github"
                />
              </svg>
            </a>
          </li>
           
          <li>
            <a
              href="https://www.facebook.com/etalab"
            >
              <svg
                class="icon icon-fb"
              >
                <use
                  xlink:href="#icon-fb"
                />
              </svg>
            </a>
          </li>
           
          <li>
            <a
              href="mailto:info@data.gouv.fr"
            >
              <svg
                class="icon icon-mail"
              >
                <use
                  xlink:href="#icon-mail"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
       
      <ul
        class="footer__links"
      >
        <li>
          <h2
            class="fit-mobile"
          >
            <a
              href="https://entreprise.data.gouv.fr/"
            >
              entreprise.data.gouv.fr
            </a>
          </h2>
        </li>
         
        <li>
          <a
            href="https://www.data.gouv.fr/fr/datasets/base-sirene-des-entreprises-et-de-leurs-etablissements-siren-siret/"
          >
            Données de référence
          </a>
        </li>
         
        <li>
          <a>
            Mentions légales
          </a>
        </li>
         
        <li>
          <a
            href="https://github.com/etalab/sirene_as_api"
          >
            API
          </a>
        </li>
      </ul>
    </div>
  </footer>
</div>
`;
